packages/mobile/src/screens/new-item/components/new-auction-form/new-auction-form.tsx

```diff
@@ -54,33 +58,52 @@ const NewAuctionForm: FC<Props> = ({ personalInfo }) => {
   const isLoading = dataStatusProducts === DataStatus.PENDING;
   const formattedCategories =
     categories && categories.length ? categoryForDropdown(categories) : null;
+  const phone = personalInfo.phone?.replace(/\s/g, '').slice(4);
+
+  const { control, errors, handleSubmit, setValue } =
+    useAppForm<ICreateAuction>({
+      defaultValues: {
+        category: '',
+        title: '',
+        description: '',
+        condition: '',
+        recommendedPriceCurrency: t('common:currency.UAH'),
+        recommendedPrice: 0,
+        minimalBidCurrency: t('common:currency.UAH'),
+        minimalBid: 0,
+        endDate: '',
+        country:
+          personalInfo.userAddress?.country || t('make_a_post.DEFAULT_COUNTRY'),
+        city: personalInfo.userAddress?.city || '',
+        phone: phone || '',
+      },
+      validationSchema: productsAuctionSchema,
+    });
 
-  const { control, errors, handleSubmit } = useAppForm<ICreateAuction>({
-    defaultValues: {
-      category: '',
-      title: '',
-      description: '',
-      condition: '',
-      recommendedPriceCurrency: t('common:currency.UAH'),
-      recommendedPrice: 0,
-      minimalBidCurrency: t('common:currency.UAH'),
-      minimalBid: 0,
-      endDate: '',
-      country:
-        personalInfo.userAddress?.country || t('make_a_post.DEFAULT_COUNTRY'),
-      city: personalInfo.userAddress?.city || '',
-      phone: personalInfo.phone?.replace(/\s/g, '').slice(4) || '',
-    },
-    validationSchema: productsAuctionSchema,
-  });
   const [images, setImages] = useState<Asset[]>([]);
+  const [hiddenPhone, setHiddenPhone] = useState(Boolean(false));
+  const isFirstRender = useRef(true);
 
   useEffect(() => {
     if (Object.keys(errors).length > 0) {
       notification.error(t('errors.CORRECTLY_FILLED'));
     }
   }, [errors]);
 
+  useEffect(() => {
+    if (isFirstRender.current) {
+      isFirstRender.current = false;
+
+      return;
+    }
+
+    if (hiddenPhone) {
+      setValue('phone', '');
+    } else {
+      setValue('phone', phone || '');
+    }
+  }, [hiddenPhone]);
+
```

Comment:
why do we need this?

Reply:
we need this because there is no field on the backend to hide the number and we send an empty number if there is a checkmark